src/EntityFramework/Core/Common/DbProviderServices.cs

```diff
@@ -863,5 +866,16 @@ public virtual IEnumerable<object> GetServices(Type type, object key)
         {
             return _resolvers.GetServices(type, key);
         }
+
+        /// <summary>
+        /// By default expression like .Select(x => NewProperty = func(x.Property)).Where(x => x.NewProperty == ...) are simplified to avoid nested SELECT
+        /// In some cases, simplifing query with UDFs could caused to suboptimal plans due to calling UDF twice. Also some SQL functions aren't allow in WHERE clause.
+        /// Disabling that behavior
+        /// </summary>
+        public static bool DisableFilterSimplificationForCustomFunctions
```

Comment:
Name suggestion: `DisableFilterOverProjectionSimplificationForCustomFunctions`

Reply:
If the behavior was provider-specific, the provider's services class would be an ok option. It isn't really great, but at least there is precedent of us putting things there, and that is why I suggested it before.  

But now I understand that it isn't going to be provider specific. Having it on the base DbProvidderServices class feels very strange. I am wondering if we can add it as an instance flag on to DbConfiguration. There isn't much of a precedent (everything there is service based and not flags), but feels correct also from the perspective of:

- Query caching: you shouldn't be able to mutate the flag once EF is initialized. I can see it being convenient, but the behavior becomes unpredictable.
- DbConfiguration is our official code-based configuration story

@ajcvickers any thoughts?

Reply:
@divega Did you mean DbContextConfiguration? I found that DbContextConfiguration has property UseDatabaseNullSemantics that passed down to PlanCompiler that uses it to decide that to apply/not apply some transformation rules (https://github.com/aspnet/EntityFramework6/blob/0ab1a198ee255ed7993b8657693d88bf8c4b7e77/src/EntityFramework/Core/Query/PlanCompiler/PlanCompiler.cs#L356) and Plan caching uses that property (it's renamed a bit) to decide that plan was cached or not (https://github.com/aspnet/EntityFramework6/blob/0ab1a198ee255ed7993b8657693d88bf8c4b7e77/src/EntityFramework/Core/Objects/ELinq/ELinqQueryState.cs#L101) It looks like pattern for transformation configuration

Reply:
I implemented the opt-in property at DbContextConfiguration. Please, take a look and let me know what do you think.